modules/core/src/test/java/org/apache/ignite/internal/processors/performancestatistics/PerformanceStatisticsSelfTest.java

```diff
@@ -0,0 +1,305 @@
+            throws EntryProcessorException {
+            return null;
+        }
+    };
+
+    /** Test cache entry processor. */
+    private static final CacheEntryProcessor<Object, Object, Object> CACHE_ENTRY_PROC =
+        new CacheEntryProcessor<Object, Object, Object>() {
+        @Override public Object process(MutableEntry<Object, Object> entry, Object... arguments)
+            throws EntryProcessorException {
+            return null;
+        }
+    };
+
+    /** Cache entry count. */
+    private static final int ENTRY_COUNT = 100;
+
+    /** Load node to run operations from. */
+    @Parameterized.Parameter
+    public LoadNode loadNode;
+
+    /** @return Test parameters. */
+    @Parameterized.Parameters(name = "loadNode={0}")
+    public static Collection<?> parameters() {
+        return Arrays.asList(new Object[][] {{SERVER}, {CLIENT}});
+    }
+
+    /** Ignite. */
+    private static IgniteEx srv;
+
+    /** Ignite node to run load from. */
+    private static IgniteEx node;
+
+    /** Test cache. */
+    private static IgniteCache<Object, Object> cache;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setCacheConfiguration(defaultCacheConfiguration());
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        srv = startGrid(0);
+
+        IgniteEx client = startClientGrid(1);
+
+        node = loadNode == SERVER ? srv : client;
+
+        cache = node.cache(DEFAULT_CACHE_NAME);
+
+        for (int i = 0; i < ENTRY_COUNT; i++)
+            cache.put(i, i);
+    }
+
+    /** @throws Exception If failed. */
+    @Test
+    public void testCompute() throws Exception {
+        String testTaskName = "testTask";
+        int executions = 5;
+
+        startCollectStatistics();
+
+        IgniteRunnable task = new IgniteRunnable() {
+            @Override public void run() {
+                // No-op.
+            }
+        };
+
+        for (int i = 0; i < executions; i++)
+            node.compute().withName(testTaskName).run(task);
+
+        HashMap<IgniteUuid, Integer> sessions = new HashMap<>();
+        AtomicInteger tasks = new AtomicInteger();
+        AtomicInteger jobs = new AtomicInteger();
+
+        stopCollectStatisticsAndRead(new TestHandler() {
+            @Override public void task(UUID nodeId, IgniteUuid sesId, String taskName, long startTime, long duration,
+                int affPartId) {
+                sessions.compute(sesId, (uuid, cnt) -> cnt == null ? 1 : ++cnt);
+
+                tasks.incrementAndGet();
+
+                assertEquals(node.context().localNodeId(), nodeId);
+                assertEquals(testTaskName, taskName);
+                assertTrue(startTime > 0);
```

Comment:
Let's store `System.currentTimeMillis` before running jobs and here check.
`assertTrue(startTime >= jobsStartTime);`

Reply:
Done